File: backend/app/api/v1/endpoints/expert_reviews.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.core.database import get_db
from app.core.deps import require_roles
from app.models.enums import UserRole
from app.models.user import User
from app.schemas.review import ExpertReviewAction, ExpertReviewActionResult, ExpertReviewDetail, ExpertReviewQueueItem
from app.services.expert_review_service import ExpertReviewService

router = APIRouter(prefix="/expert-reviews", tags=["expert-reviews"])


@router.get("", response_model=list[ExpertReviewQueueItem])
def list_review_queue(
    _: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    return ExpertReviewService(db).list_queue()


@router.get("/{prescription_id}", response_model=ExpertReviewDetail)
def get_review_detail(
    prescription_id: int,
    _: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    return ExpertReviewService(db).detail(prescription_id)


@router.post("/{prescription_id}/approve", response_model=ExpertReviewActionResult)
def approve_prescription(
    prescription_id: int,
    payload: ExpertReviewAction,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    review, prescription = ExpertReviewService(db).approve(
        prescription_id=prescription_id,
        expert_id=current_user.id,
        review_comment=payload.review_comment,
        edited_prescription=payload.edited_prescription,
    )
    return {"status": "APPROVED", "review": review, "prescription": prescription}


@router.post("/{prescription_id}/reject", response_model=ExpertReviewActionResult)
def reject_prescription(
    prescription_id: int,
    payload: ExpertReviewAction,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    review, prescription = ExpertReviewService(db).reject(
        prescription_id=prescription_id,
        expert_id=current_user.id,
        review_comment=payload.review_comment,
    )
    return {"status": "REJECTED", "review": review, "prescription": prescription}


@router.post("/{prescription_id}/refer", response_model=ExpertReviewActionResult)
def refer_prescription(
    prescription_id: int,
    payload: ExpertReviewAction,
    current_user: User = Depends(require_roles(UserRole.EXPERT, UserRole.ADMIN, UserRole.ORG_ADMIN)),
    db: Session = Depends(get_db),
):
    review, prescription = ExpertReviewService(db).refer(
        prescription_id=prescription_id,
        expert_id=current_user.id,
        review_comment=payload.review_comment,
    )
    return {"status": "REFERRED", "review": review, "prescription": prescription}

File: frontend/src/api/expertReviews.ts
import { apiClient } from "./client";

export interface ReviewQueueItem {
  prescription_id: number;
  user_id: number;
  risk_level: string;
  status: string;
  version: number;
  created_at: string;
  review_id: number | null;
}

export interface ReviewDetail {
  prescription: {
    id: number;
    risk_level: string;
    status: string;
    fitt_vp: Record<string, unknown> | null;
    precautions: string[];
    contraindications: string[];
    evidence_refs: Array<Record<string, unknown>>;
  };
  review: { id: number; status: string };
  health_snapshot: Record<string, Record<string, unknown> | null>;
  risk_rules: Array<Record<string, unknown>>;
  evidence_refs: Array<Record<string, unknown>>;
  template: Record<string, unknown> | null;
  candidate_actions: Array<Record<string, unknown>>;
}

export function listReviewQueue() {
  return apiClient.get<ReviewQueueItem[]>("/expert-reviews").then((response) => response.data);
}

export function getReviewDetail(prescriptionId: number) {
  return apiClient.get<ReviewDetail>(`/expert-reviews/${prescriptionId}`).then((response) => response.data);
}

export function approvePrescription(prescriptionId: number, payload: Record<string, unknown>) {
  return apiClient.post(`/expert-reviews/${prescriptionId}/approve`, payload).then((response) => response.data);
}

export function rejectPrescription(prescriptionId: number, payload: Record<string, unknown>) {
  return apiClient.post(`/expert-reviews/${prescriptionId}/reject`, payload).then((response) => response.data);
}

export function referPrescription(prescriptionId: number, payload: Record<string, unknown>) {
  return apiClient.post(`/expert-reviews/${prescriptionId}/refer`, payload).then((response) => response.data);
}
